model/role.go

```diff
@@ -484,11 +484,17 @@ func (r *Role) IsValidWithoutId() bool {
 
 	for _, permission := range r.Permissions {
 		permissionValidated := false
-		for _, p := range append(AllPermissions, DeprecatedPermissions...) {
-			if permission == p.Id {
-				permissionValidated = true
-				break
+		check := func(perms []*Permission, permission string) bool {
+			for _, p := range perms {
+				if permission == p.Id {
+					return true
+				}
 			}
+			return false
+		}
+		permissionValidated = check(AllPermissions, permission)
+		if !permissionValidated {
+			permissionValidated = check(DeprecatedPermissions, permission)
```

Comment:
Would it be too much to exploit short circuiting and doing:

```go
permissionValidated := check(AllPermissions, permission) ||
   check(DeprecatedPermissions, permission)
```

That's why I originally suggested the `bool` return actually but I totally understand if it's just me that finds that preferable.

Reply:
Nice. My brain cannot come up with anything more complicated than simple if conditions. :rofl: